src/Aggregation/NormalizeAggregation.php

```diff
@@ -19,6 +19,10 @@ public function __construct(string $name, ?string $bucketsPath = null, ?string $
 
         if (null !== $bucketsPath) {
```

Comment:
It seems the top part of the constructor is always shared. Made me wonder if we should extract it.

Reply:
Yes it's shared by a lot of pipeline aggregation, but there're some exceptions like `bucket_script` or `bucket_selector` which accepts an array of buckets paths.
A topic I have in mind for 8.0 version is to review the aggregations classes to separate them in three categories (as described in https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations.html) `pipeline`, `metric` and `bucket` as they have some difference (for example a pipeline aggregation can't have sub aggregations, ...)